src/ai/backend/manager/models/utils.py

```diff
@@ -550,3 +562,34 @@ async def vacuum_db(
             vacuum_sql = "VACUUM FULL" if vacuum_full else "VACUUM"
             log.info(f"Perfoming {vacuum_sql} operation...")
             await conn.exec_driver_sql(vacuum_sql)
+
+
+def get_lastest_timestamp_for_status(
+    status_history_log: list[dict[str, str]],
+    status: str,
+) -> Optional[datetime]:
+    """
+    Get the last occurrence time of the given status from the status history records.
+    If the status is not found, return None.
+    """
+
+    for item in reversed(status_history_log):
+        if item["status"] == status:
+            return dtparse(item["timestamp"])
+    return None
```

Comment:
@HyeockJinKim `reversed()` return **Iterator**, so it allows iterating from the end of the list without copying the entire list, eliminating the need to use `range()`.
But we should use `reversed()` alone, not `list(reversed(...))` which copy the entire list.